src/test/java/uk/gov/digital/ho/hocs/casework/api/BankHolidayIntegrationTest.java

```diff
@@ -0,0 +1,115 @@
+@ActiveProfiles({ "local", "integration" })
+public class BankHolidayIntegrationTest {
+
+    private final TestRestTemplate testRestTemplate = new TestRestTemplate();
+
+    @Autowired
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    @LocalServerPort
+    int port;
+
+    @Before
+    public void setUp() throws IOException {
+        File firstHolidayResponse = new File(
+                this.getClass().getClassLoader().getResource("bank_holidays_response_1.json").getFile()
+        );
+
+        File secondHolidayResponse = new File(
+                this.getClass().getClassLoader().getResource("bank_holidays_response_2.json").getFile()
+        );
+
+        MockRestServiceServer mockBankHolidayApi = buildMockService(restTemplate);
+
+        mockBankHolidayApi
+                .expect(requestTo("https://www.gov.uk/bank-holidays.json"))
+                .andExpect(method(GET))
+                .andRespond(withSuccess(Files.readString(firstHolidayResponse.toPath()), MediaType.APPLICATION_JSON));
+
+        mockBankHolidayApi
+                .expect(requestTo("https://www.gov.uk/bank-holidays.json"))
+                .andExpect(method(GET))
+                .andRespond(withSuccess(Files.readString(secondHolidayResponse.toPath()), MediaType.APPLICATION_JSON));
+    }
+
+    @Test
+    public void shouldUpdateBankHolidaysAndReturn200_thenAddNewBankHolidaysAndReturn200() {
+
+        ResponseEntity<String> firstResponse = testRestTemplate.exchange(
+                getBasePath() + "/bankHoliday/refresh",
+                GET,
+                new HttpEntity<>(createValidAuthHeaders()),
+                String.class
+        );
+
+        assertThat(firstResponse.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(firstResponse.getBody()).isNotNull();
+        assertThat(firstResponse.getBody()).isEqualTo("Success: Added 1 bank holidays");
+
+        ResponseEntity<String> secondResponse = testRestTemplate.exchange(
+                getBasePath() + "/bankHoliday/refresh",
+                GET,
+                new HttpEntity<>(createValidAuthHeaders()),
+                String.class
+        );
+
+        assertThat(secondResponse.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(secondResponse.getBody()).isNotNull();
+        assertThat(secondResponse.getBody()).isEqualTo("Success: Added 3 bank holidays");
+    }
+
+    // -------- HELPERS ----------
+
+    private HttpHeaders createValidAuthHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.add("X-Auth-Groups", "/RERERCIiIiIiIiIiIiIiIg");
+        headers.add("X-Auth-Userid", "a.person@digital.homeoffice.gov.uk");
+        headers.add("X-Correlation-Id", "1");
+        return headers;
+    }
+
+    private String getBasePath() {
```

Comment:
Probably out of scope, I see this throughout the integration test files, might be a good shout at some point having a base integration test file.